tket2-hseries/src/lazify_measure.rs

```diff
@@ -125,7 +125,14 @@ impl LazifyMeasureRewrite {
         Self::check_signature(node, QSystemOp::LazyMeasure, hugr.get_optype(node))?;
 
         let subgraph = SiblingSubgraph::from_node(node, &hugr);
-        let uses = hugr.linked_inputs(node, 0).collect_vec();
+        let uses = {
+            // SimpleReplacement adds edges in a nondeterministic order.  This
+            // results in linked_inputs returning items in a nondeterministic
+            // order. We sort them here to restore determinism.
+            let mut v = hugr.linked_inputs(node, 0).collect_vec();
+            v.sort();
+            v
+        };
```

Comment:
A bit simpler using `Itertools::sorted`
```suggestion
        // SimpleReplacement adds edges in a nondeterministic order.  This
        // results in linked_inputs returning items in a nondeterministic
        // order. We sort them here to restore determinism.
        let uses = hugr.linked_inputs(node, 0).sorted().collect_vec();
```